src/integration-test/java/com/mongodb/jdbc/integration/MongoSQLIntegrationTest.java

```diff
@@ -55,4 +80,68 @@ public void loadTestConfigs() throws IOException {
         }
         return dynamicTests;
     }
+
+    @Test
+    public void testTypesOther() throws SQLException, IOException, IllegalAccessException {
+        DataLoader loader = new DataLoader(TEST_DATA_OTHER_DIRECTORY);
+        loader.dropCollections();
+
+        Map<String, Object> row = new LinkedHashMap<>();
+        ObjectId objID = new ObjectId("000000000000000000000001");
+        BsonObjectId objectID = new BsonObjectId(objID);
+        BsonDbPointer dbPointer = new BsonDbPointer("namespace", objID);
+        BsonJavaScript javaScript = new BsonJavaScript("function(){ }");
+        BsonRegularExpression regularExpression = new BsonRegularExpression("a(bc)*");
+        BsonSymbol symbol = new BsonSymbol("symbol");
+        BsonTimestamp timestamp = new BsonTimestamp(100);
+
+        row.put("_id", objectID);
+        row.put("a", dbPointer);
+        row.put("b", javaScript);
+        row.put("c", new BsonMaxKey());
+        row.put("d", new BsonMinKey());
+        row.put("e", regularExpression);
+        row.put("f", symbol);
+        row.put("g", timestamp);
+        /*
+        BsonArray array = new BsonArray();
```

Comment:
Troubleshooting the BsonArray type still, it is returning the error:
"failed getting result set schema: failed to convert BSON document to json_schema::Schema: data did not match any variant of untagged enum BsonType"

Reply:
Added skip reason for this: SQL-697
The problem looks related to how the arrays schema has an items field that mongosql-rs is expecting to be a single schema but is an array.